### DISCUSSION TITLE
Integration of shadcn-vue in NiceGUI

### DISCUSSION THREAD

**Discussion by covapps** [zauberzeug/nicegui#2534] [Ideas / Feature Requests]
For a different theming and components it's to be considered to add shadcn.

https://www.shadcn-vue.com

It's widely used and in the web framework world used as a cookbook but it would be nice to have it out of the box in NiceGUI

**Comment by evnchn** [comment]
Unfortunately not quite. Component library and Theme library are two distinct concepts, though often bundled in one. 

Since we use Quasar for the components library, we currently are forced to use its theme library. 

To use `shadcn-vue`'s theme library, we'd need to swap over our components library. 

_Or do we?_ In https://github.com/zauberzeug/nicegui/discussions/5608 we see https://basecoatui.com/ stand out as 99% a theme library with 1% component library left. 

https://github.com/hunvreus/basecoat/pull/130 is an attempt to extract that 99% and toss it onto Quasar. Either that gets merged, or we have to maintain a fork.
